### ISSUE TITLE
DataTable - Backwards keyboard traversal from search field doesn't work

### ISSUE THREAD

**Issue by jendowns** [carbon-design-system/carbon#3762]
## Detailed description

You should be able to tab backwards (`shift` + `tab`) within the batch actions of the `DataTable`, but your focus will get stuck in an *empty* search field and you cannot go backwards past it. (Please see steps below to reproduce.)

**NOTE:** This situation only applies to an empty search field, as far as I can tell.

## Steps to reproduce the issue

1. Go here: http://react.carbondesignsystem.com/?path=/story/datatable--with-batch-actions
2. Start to tab through the `DataTable`. Don't type anything in the search field, just tab through to the "settings" or "Add new" buttons.
3. Use `shift` + `tab` to tab backwards
4. Note that you can't traverse backwards past the search field. The empty search field stops backwards keyboard traversal entirely.
